### DISCUSSION TITLE
How to deconstruct data from useFetch?

### DISCUSSION THREAD

**Discussion by crotanite** [nuxt/nuxt#16777] [Questions]
```
------------------------------
- Operating System: `Linux`
- Node Version:     `v16.15.1`
- Nuxt Version:     `3.0.0-rc.8-27681070.ad63817`
- Package Manager:  `unknown`
- Builder:          `vite`
- User Config:      `auth`, `axios`, `build`, `components`, `css`, `intlify`, `modules`, `proxy`, `runtimeConfig`, `vite`
- Runtime Modules:  `@nuxtjs-alt/auth@1.2.12`, `@nuxtjs-alt/axios@1.0.16`, `@pinia/nuxt@0.4.1`, `@nuxtjs/tailwindcss@5.3.2`, `@nuxtjs-alt/proxy@1.3.1`
- Build Modules:    `-`
------------------------------
```

I can deconstruct the data from useFetch with the following:
```
const { pending, error, data: { value: { data: discussion = ref(null) } } } = await useFetch('/route')
```

This works when navigating to the page, but _not_ on page refresh. 
How can I make this work on a hard refresh as well?

**Comment by sevillaarvin** [comment]
Perhaps you don’t have to do it this way? It’s not very readable.

In any case I don’t think you want `discussion = ref(null)` because `ref(null)` will only be used if discussion is null. 

How about defining it separately: `let discussion = ref(null)` then destructure it `data: { value: { data: discussion } } }`
